### PR TITLE
server, schedulers: introduce label property and label scheduler.

### PR DESCRIPTION
**Pull request by disksing**
To support forbidding stores with some labels to have any region leaders.

### CODE REVIEW

File: server/option.go [hunk 1]
```diff
@@ -19,15 +19,17 @@ import (
...
 
 // scheduleOption is a wrapper to access the configuration safely.
 type scheduleOption struct {
-	v   atomic.Value
-	rep *Replication
-	ns  map[string]*namespaceOption
+	v             atomic.Value
+	rep           *Replication
+	ns            map[string]*namespaceOption
+	labelProperty LabelPropertyConfig
```
Comment (Connor1996) [inline server/option.go]: no need to persist?

Comment (disksing) [reply server/option.go]: Working on it. Maybe in separated PR.

**Comment by nolouch** [comment]
should also `CreateMovePeerOperator` need to consider the reject store?

**Comment by disksing** [comment]
@nolouch Fixed the issue. PTAL.

File: server/schedule/operator.go [hunk 2]
```diff
@@ -262,38 +262,32 @@ func (o *Operator) History() []OperatorHistory {
...
+func CreateMovePeerOperator(desc string, cluster Cluster, region *core.RegionInfo, kind OperatorKind, oldStore, newStore uint64, peerID uint64) *Operator {
+	kind, steps := removePeerSteps(cluster, region, oldStore)
+	steps = append([]OperatorStep{AddPeer{ToStore: newStore, PeerID: peerID}}, steps...)
+	return NewOperator(desc, region.GetId(), kind|OpRegion, steps...)
+}
+
+// removePeerSteps returns the steps to safely remove a peer. It prevents removing leader by transfer its leadership first.
+func removePeerSteps(cluster Cluster, region *core.RegionInfo, storeID uint64) (kind OperatorKind, steps []OperatorStep) {
+	if region.Leader != nil && region.Leader.GetStoreId() == storeID {
+		for id := range region.GetFollowers() {
```
Comment (Connor1996) [inline server/schedule/operator.go]: what if all followers reject leader?

Comment (disksing) [reply server/schedule/operator.go]: Then we have to remove the leader, which will cause inevitable leader election.

**Pull request merged**